syntax = "proto3";

package dr526;

option cc_generic_services = true;

message resultCode
{
    int32 errCode = 1;
    bytes errmsg = 2;
}

message loginRequest
{
    bytes name = 1;
    bytes pwd = 2;
}

message loginResponse
{
    resultCode result = 1;
    bool success = 2;
}

message registeRequest
{
    uint32 id = 1;
    bytes name = 2;
    bytes pwd = 3;
}

message registeResponse
{
    resultCode result = 1;
    bool success = 2;
}

service userServiceRpc
{
    rpc login(loginRequest) returns(loginResponse);
    rpc registe(registeRequest) returns(registeResponse);
}